### PR TITLE
[LENS-532] Range Slider

### PR DESCRIPTION
**Pull request by dbchristopher**
### :sparkles: Changes

- **Thank you for your contribution to Looker Components!**
- To ensure a successful review, please double check our checklist of requirements below.
- Also including a short description of changes and relevant screenshots will help us more quickly understand the ultimate goal of your work (🗑 replace this introduction with your summary of changes).

### :white_check_mark: Requirements

- [x] Includes test coverage for all changes
- [x] Build and tests are passing
- [x] Update documentation
- [x] Updated CHANGELOG
- [ ] Checked for i18n impacts
- [x] Checked for a11y impacts
- [ ] PR is ideally < ~400LOC

### CODE REVIEW

File: packages/components/package.json [hunk 1]
```diff
@@ -42,6 +42,7 @@
     "react-resize-detector": "^4.2.3",
     "react-transition-group": "^4.3.0",
     "@types/styled-system": "^5.1.9",
+    "resize-observer-polyfill": "^1.5.1",
```
Comment (dbchristopher) [inline packages/components/package.json:45]: resize observer can notify you if the element dimensions change independently of a window resize event. super cool

File: packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx [hunk 2]
```diff
@@ -0,0 +1,464 @@
...
+          onKeyDown={handleKeyboardNav}
+          ref={minThumbRef}
+          disabled={disabled}
+          aria-label="Minimum Value"
+          role="slider"
+          aria-valuemin={min}
+          aria-valuemax={max}
+          aria-disabled={disabled}
+          aria-valuenow={value[0]}
+          aria-labelledBy={ariaLabelledBy}
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx:365]: I was really happy to discover these aria attributes were available for this kind of ui pattern

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 3]
```diff
@@ -224,20 +199,6 @@ const SliderInput = styled.input.attrs({ type: 'range' })<SliderInputProps>`
...
-
-    &::-moz-range-thumb {
-      transform: scale3d(1.15, 1.15, 1);
-    }
-
-    &::-ms-thumb {
-      transform: scale3d(1.15, 1.15, 1);
-    }
-  }
-
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx:240]: I updated the single-value Slider styles to match the new range-slider focus state

File: packages/components/src/Form/Inputs/Slider/Slider.tsx [hunk 4]
```diff
@@ -27,18 +27,15 @@
 import React, { forwardRef, Ref, SyntheticEvent, useState } from 'react'
 import isFunction from 'lodash/isFunction'
 import styled, { css } from 'styled-components'
-import { rgba } from 'polished'
 import { reset, space, SpaceProps } from '@looker/design-tokens'
-import { WidthProps, width, fontSize, FontSizeProps } from 'styled-system'
+import { WidthProps, width, FontSizeProps } from 'styled-system'
 
 import { InputProps } from '../InputProps'
-import { SliderSizeProps, sliderSize } from './slider_sizes'
```
Comment (dbchristopher) [inline packages/components/src/Form/Inputs/Slider/Slider.tsx:35]: Deleted the multiple slider sizes. Jared wasn't into that in the first place, and I'd rather have fewer variants of this to maintain anyways

File: packages/playground/src/Form/RangeSliderDemo.tsx [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+            <RangeSlider min={100} max={1000} step={50} />
+          </CardContent>
+          <CardContent p="xxlarge">
+            <Heading>Min: 52, Max: 111, Step: 17</Heading>
+            <RangeSlider min={52} max={111} step={17} />
+          </CardContent>
+          <CardContent p="xxlarge">
+            <Heading>Floating point range</Heading>
+            <RangeSlider min={0.1} max={1.1} step={0.1} />
+          </CardContent>
```
Comment (dbchristopher) [inline packages/playground/src/Form/RangeSliderDemo.tsx:63]: This actually works better than I expected with no additional effort on my part. Still, I hope people just use this with round integers 😄

**Comment by dbchristopher** [comment]
@mdodgelooker this is ready for a review whenever you have time next week! 🎉

File: packages/playground/src/Form/RangeSliderDemo.tsx (hunk 5 above)
Comment (mdodgelooker) [reply packages/playground/src/Form/RangeSliderDemo.tsx:63]: Not bad! Good ol' floating point math gives us some funny numbers but I believe the core product only allows integers, so it's moot: 
image

File: packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx [hunk 6]
```diff
@@ -0,0 +1,464 @@
...
+   */
+  const [value, setValue] = useState(sort(boundedValue))
+  const [containerRef, setContainerRef] = useState<HTMLElement | null>(null)
+  const [focusedThumb, setFocusedThumb] = useState<ThumbIndices>()
+
+  const callbackRef = useCallback((node) => {
+    if (node !== null) {
+      setContainerRef(node)
+    }
+  }, [])
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx:189]: At one point I created `utils/useCallbackRef.ts` for this kind of use case, but later I realized (at least I thought I did...) that you can just directly assign the state setter like `ref={setContainerRef}`.

File: packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx [hunk 7]
```diff
@@ -0,0 +1,464 @@
...
+        newPoint,
+        maintainFocus ? focusedThumb : undefined
+      )
+      focusChangedPoint(newValue, newPoint)
+      setValue(newValue)
+    }
+  }
+
+  const handleMouseDown = partial(handleMouseEvent, false)
+  const handleMouseDrag = partial(handleMouseEvent, true)
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx:283]: Clever! I'll have to steal this.

File: packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx [hunk 8]
```diff
@@ -0,0 +1,464 @@
...
+  focusedIndex?: ThumbIndices
+) => {
+  const indexToReplace =
+    focusedIndex === undefined
+      ? findClosestIndex(value, newPoint)
+      : focusedIndex
+
+  const newValue = Object.assign([], value, { [indexToReplace]: newPoint })
+  return sort(newValue)
+}
```
Comment (mdodgelooker) [inline packages/components/src/Form/Inputs/RangeSlider/RangeSlider.tsx:100]: Awesome!

**Comment by dbchristopher** [comment]
thanks for the review @mdodgelooker! I implemented your feedback and this should be ready for another look

**Pull request merged**